tests/src/rules/extensions.js

```diff
@@ -594,6 +614,35 @@ ruleTester.run('extensions', rule, {
         },
       ],
     }),
+
+    // TODO: properly ignore packages resolved via relative imports
+    test({
+      code: [
+        'import * as test from "."',
+      ].join('\n'),
+      filename: testFilePath('./internal-modules/test.js'),
+      options: [ 'ignorePackages' ],
+      errors: [
+        {
+          message: 'Missing file extension for "."',
+          line: 1,
+        },
+      ],
+    }),
+    // TODO: properly ignore packages resolved via relative imports
+    test({
+      code: [
+        'import * as test from ".."',
+      ].join('\n'),
+      filename: testFilePath('./internal-modules/plugins/plugin.js'),
+      options: [ 'ignorePackages' ],
```

Comment:
I'm not sure about whether these should be `always`. I think with `ignorePackages` it should definitely not error, so it makes for a less ambiguous test case.

In the always + relative case it's a bit more ambiguous to me, since if we assume the relative path is in your project, you might have this set to `always` because you always want someone to choose a file (within the project).

Reply:
I'm not sure what to do here, tbh.

Reply:
I'm going to merge this, and please file followup issues and/or PRs :-)